imap/imap.c

```diff
@@ -1974,13 +1974,9 @@ int imap_login(struct ImapAccountData *adata)
 /**
  * imap_mbox_open - Implements MxOps::mbox_open()
  */
-static int imap_mbox_open(struct Context *ctx)
+static int imap_mbox_open(struct Mailbox *m, struct Context *ctx)
 {
-  if (!ctx || !ctx->mailbox)
-    return -1;
-
-  struct Mailbox *m = ctx->mailbox;
-  if (!m->account)
+  if (!m || !m->account)
```

Comment:
Is this guard statement necessary given `mx_mbox_open(...)`'s `NULL` checking and initialisation strategy?